packages/filter/src/__tests__/acceptance/query-builder.acceptance.ts

```diff
@@ -301,7 +301,17 @@ describe('FilterBuilder', () => {
       },
     });
   });
-
+  it('builds a filter object with array', () => {
+    const filterBuilder = new FilterBuilder();
+    filterBuilder.fields(['a', 'b']);
+    const filter = filterBuilder.build();
+    expect(filter).to.eql({
+      fields: {
+        a: true,
+        b: true,
```

Comment:
Is it necessary to convert `fields` from an array to an object? I believe juggler and connectors should support both object and array format, so I don't think we need to implement this conversion in `FilterBuilder`.

We have test cases to verify that all connectors support both formats, see [`loopback-datasource-juggler/test/basic-querying.test.js`](https://github.com/strongloop/loopback-datasource-juggler/blob/8031e54fb4a616a54323d665833d5a7e255ad202/test/basic-querying.test.js#L839-L847). The base SQL connector seems to support both array and object format, see e.g. [here](https://github.com/strongloop/loopback-connector/blob/576e956dc1164888e29de570ce7dbe2b4b4af886/lib/sql.js#L1294-L1299) and so does our MongoDB connector (see [here](https://github.com/strongloop/loopback-connector-mongodb/blob/4551cc0428a6954091981eb8e4861e7c1c9b37d8/lib/mongodb.js#L156-L168)).

I believe that internally, juggler will convert `fields` from an object into an array anyways, see [`ModelUtils._normalize()`](https://github.com/strongloop/loopback-datasource-juggler/blob/8031e54fb4a616a54323d665833d5a7e255ad202/lib/model-utils.js#L173-L177)

My conclusion is that `FilterBuilder` should keep not modify `filter.fields` value and keep it as an object or as an array, depending on what was provided by the user.

@raymondfeng @jannyHou @hacksparrow what do you think?

Reply:
Thank you for a detailed explanation! Based on your arguments, I am fine to keep representing fields as an object internally inside `FilterBuilder`. (Unless other maintainers disagree.)